craft_parts/plugins/base.py

```diff
@@ -62,6 +62,11 @@ def get_build_environment(self) -> Dict[str, str]:
 
     @property
     def out_of_source_build(self) -> bool:
```

Comment:
Should I remove the `out_of_source_build` property?

The only place it is used across all craft libraries and *craft applications is the `ros` plugin in snapcraft.